packages/ove-app-networks/src/client/common/networks.js

```diff
@@ -13,6 +13,235 @@ $(function () {
+            }
+        };
+        const g = window.ove.context.sigma.graph;
+        reset(g.nodes(), true);
+        reset(g.edges(), false);
+    })();
+
+    // The decision for making edge/node operations are dependent on the presence of the
+    // corresponding filter
+    const nodeFilter = getFromMessage(message, 'node', '$filter');
+    const edgeFilter = getFromMessage(message, 'edge', '$filter');
+
+    let f = (new sigma.plugins.filter(window.ove.context.sigma)).undo();
+    switch (message.operation) {
+        case Constants.Operation.SEARCH:
+            if (nodeFilter) {
+                log.debug('Filtering nodes using filter:', nodeFilter);
+                f = f.nodesBy(function (n) {
+                    return evaluate(n, nodeFilter);
+                });
+            }
+            if (edgeFilter) {
+                log.debug('Filtering edges using filter:', edgeFilter);
+                f.edgesBy(function (n) {
+                    return evaluate(n, edgeFilter);
+                });
+            }
+            f.apply();
+            break;
+        case Constants.Operation.COLOR:
+            if (nodeFilter) {
+                const nodeColor = getFromMessage(message, 'node', 'color');
+                log.debug('Applying color:', nodeColor, 'on all nodes matching filter:', nodeFilter);
+                f = f.nodesBy(function (n) {
+                    if (evaluate(n, nodeFilter)) {
+                        n.color = nodeColor;
+                    }
+                    return true;
+                });
+            }
+            if (edgeFilter) {
+                const edgeColor = getFromMessage(message, 'edge', 'color');
+                log.debug('Applying color:', edgeColor, 'on all edges matching filter:', edgeFilter);
+                f.edgesBy(function (n) {
+                    if (evaluate(n, edgeFilter)) {
+                        n.color = edgeColor;
+                    }
+                    return true;
+                });
+            }
+            f.apply();
+            break;
+        case Constants.Operation.LABEL:
+            if (nodeFilter) {
+                const nodeLabel = getFromMessage(message, 'node', 'label');
+                log.debug('Applying label:', nodeLabel, 'on all nodes matching filter:', nodeFilter);
+                f = f.nodesBy(function (n) {
+                    if (evaluate(n, nodeFilter)) {
+                        n.label = getFromElement(n, nodeLabel);
+                    }
+                    return true;
+                });
+            } else {
+                const nodeLabel = getFromMessage(message, 'node', 'label');
+                log.debug('Applying label:', nodeLabel, 'on all nodes');
+                f = f.nodesBy(function (n) {
+                    n.label = getFromElement(n, nodeLabel);
+                    return true;
+                });
+            }
+            f.apply();
+            break;
+        case Constants.Operation.NEIGHBORS_OF:
```

Comment:
https://grammarist.com/spelling/neighbor-neighbour/ 
may we not be American please ;-)

Reply:
@davidbirchwork, please have a look at https://github.com/jacomyal/sigma.js/tree/master/plugins/sigma.plugins.filter. We may not be American, but we cannot have two spellings for the same operation.